tensorflow/compiler/jit/kernels/xla_ops.cc

```diff
@@ -381,6 +376,20 @@ void XlaCompileOp::Compute(OpKernelContext* ctx) {
     mutex_lock guard(cannot_compile_cluster_mu_);
     cannot_compile_cluster = cannot_compile_cluster_;
   }
+  // If must_compile_ is true, there is no fallback path and therefore
+  // async and lazy must be false. If must_compile_ is false, and async
+  // compilation is enabled, async is true, and lazy is false. Otherwise
+  // lazy compilation is true.
+  bool async = !must_compile_ &&
+               GetXlaOpsCommonFlags().tf_xla_async_compilation;
+  // Possible future work:
+  //    disable async for small clusters.
+  //    disable async for cluster that have short compile time.
+  bool lazy = async ? false : !must_compile_;
+  XlaCompilationCache::CompileMode cmode =
+    lazy  ? XlaCompilationCache::CompileMode::kLazy :
+    async ? XlaCompilationCache::CompileMode::kAsync :
+            XlaCompilationCache::CompileMode::kStrict;
```

Comment:
IMO this will be easier to read and self documenting if we write it as:

```
XlaCompilationCache::CompileMode compile_mode = [&] {
  if (must_compile_) { return kStrict; }
  return GetXlaOpsCommonFlags().tf_xla_async_compilation ? kAsync : kLazy;
}();
```

Reply:
Passing additional to CompileImpl requires many additional functions tot changes as well. Capture by value is cleaner.
Added a POD-type instead of a class with a operator() to explicitly capture the required variables, added additional comments for clarity, and most importantly, removed capture-default to prevent  additional variables to be introduced in the body of the lambda